privval/socket_test.go

```diff
@@ -82,3 +84,89 @@ func TestTCPListenerConnDeadline(t *testing.T) {
 	close(readyc)
 	<-donec
 }
+
+// testUnixAddr will attempt to obtain a platform-independent temporary file
+// name for a Unix socket
+func testUnixAddr() (string, error) {
+	f, err := ioutil.TempFile("", "tendermint-privval-test")
+	if err != nil {
+		return "", err
+	}
+	addr := f.Name()
+	f.Close()
+	os.Remove(addr)
+	return addr, nil
+}
+
+func TestUnixListenerAcceptDeadline(t *testing.T) {
+	addr, err := testUnixAddr()
+	if err != nil {
+		t.Fatal(err)
+	}
+	ln, err := net.Listen("unix", addr)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	unixLn := NewUnixListener(ln)
+	UnixListenerAcceptDeadline(time.Millisecond)(unixLn)
+	UnixListenerConnDeadline(time.Second)(unixLn)
+
+	_, err = unixLn.Accept()
+	opErr, ok := err.(*net.OpError)
+	if !ok {
+		t.Fatalf("have %v, want *net.OpError", err)
+	}
+
+	if have, want := opErr.Op, "accept"; have != want {
+		t.Errorf("have %v, want %v", have, want)
+	}
+}
+
+func TestUnixListenerConnDeadline(t *testing.T) {
+	addr, err := testUnixAddr()
+	if err != nil {
+		t.Fatal(err)
+	}
+	ln, err := net.Listen("unix", addr)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	unixLn := NewUnixListener(ln)
```

Comment:
both tcp and unix listeners should adhere to `net.Listener` interface. it means we can probably merge two tests into one (same thing we did with rpc clients https://github.com/tendermint/tendermint/blob/ec53ce359bb8f011e4dbb715da098bea08c32ded/rpc/client/rpc_test.go#L53). m?

Reply:
I've combined the tests now into an iterable set of test cases. Does this work?

Reply:
Yep, thanks